pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -91,17 +92,33 @@ func (a *v1beta1PolicyApplier) setAuthnFilterForPeerAuthn(proxyType model.NodeTy
 	p := config.Policy
 	p.Peers = []*authn_alpha.PeerAuthenticationMethod{}
 
-	effectiveMTLSMode := a.getMutualTLSModeForPort(port)
-	if effectiveMTLSMode == model.MTLSPermissive || effectiveMTLSMode == model.MTLSStrict {
-		mode := authn_alpha.MutualTls_PERMISSIVE
-		if effectiveMTLSMode == model.MTLSStrict {
-			mode = authn_alpha.MutualTls_STRICT
+	if proxyType == model.SidecarProxy {
+		effectiveMTLSMode := a.getMutualTLSModeForPort(port)
+		if effectiveMTLSMode == model.MTLSPermissive || effectiveMTLSMode == model.MTLSStrict {
+			mode := authn_alpha.MutualTls_PERMISSIVE
+			if effectiveMTLSMode == model.MTLSStrict {
+				mode = authn_alpha.MutualTls_STRICT
+			}
+			p.Peers = []*authn_alpha.PeerAuthenticationMethod{
+				{
+					Params: &authn_alpha.PeerAuthenticationMethod_Mtls{
+						Mtls: &authn_alpha.MutualTls{
+							Mode: mode,
+						},
+					},
+				},
+			}
 		}
+	} else {
+		// this is for gateway with a server whose TLS mode is ISTIO_MUTUAL
```

Comment:
You should set `SkipValidateTrustDomain: true` in this case otherwise the gateway will reject requests if not from the same trust domain.